### DISCUSSION TITLE
Consuming API examples

### DISCUSSION THREAD

**Discussion by markdav-is** [oqtane/oqtane.framework#1807] [Q&A]
We have a few projects written using Oqtane and one of the compelling reasons was the API support.  We are now getting to the point where we are starting to expose API endpoints to external services and are looking for guidance.

Q: How best to authenticate a user when calling the API?  My assumptions would be that we'd establish an API account for the consuming service and create a standard user for them.  Then they could login programmatically via the user API and get a token.  Are there any examples of this?  I'd love to be able to use a tool like postman to login and get a token, then use that token to process a request to another API and have it resolve.

Thanks in advance for any help.

**Comment by sbwalker** [comment]
@markdav-is this is the more frequently requested feature and will be the focus of the Oqtane 3.1 release... I announced this in my .NET Conf presentation last week:

![image](https://user-images.githubusercontent.com/4840590/142496550-14a84fbb-900a-44f7-9c95-3bd032a32e17.png)

**Comment by markdav-is** [comment]
awesome.  I'd love to help!

**Comment by markdav-is** [comment]
I was chatting with my business partner about identity providers now that ID4 is sunset and Duende is charging for prof installs.  If Oqtane supported for self-hosted auth scenarios, that could be one more compelling reason to use it for user management.
